e2e/test/provisioning/ProvisioningE2ETests.cs

```diff
@@ -459,10 +459,17 @@ public async Task ProvisioningDeviceClient_ValidRegistrationId_TimeSpanTimeoutRe
             {
                 return; // expected exception was thrown, so exit the test
             }
+            catch (ProvisioningTransportException ex) when (ex.InnerException is OperationCanceledException)
+            {
+                // The provisioning transport layer will typically throw an OperationCanceledException, but some clock issues
+                // may cause this test to throw the OperationCanceledException nested in a ProvisioningTransportException instead.
+                return; // expected exception was thrown, so exit the test
+            }
```

Comment:
We have a check on the cancellation token that we create from a timespan as the first line in our provisioning transport handler for HTTP, but for some reason, it can sometimes pass that check even with Timespan.Zero. The next check on the timespan will throw an OperationCanceledException, but nested in a ProvisioningTransportException.

Reply:
this is interesting, do you have a link to the code that you're referring to?

Reply:
https://github.com/Azure/azure-amqp/blob/0c4cf8a15b7df2cbe155e59c26ea23cfb6d3f652/src/Transport/TcpTransportInitiator.cs#L38

Reply:
Can we add a check in our SDK to ensure TimeSpan.Zero always throws an OperationCancelledException. The reason I say this is, while these layered checks might be ok for our tests, they'll make an uncomfortable usage experience for our customers.

Reply:
I can, but do you expect users to call these APIs with a timespan of 0?

Reply:
I get that a user wouldn't want to make the same catch statements that I made for this test, but adding a check for Timespan.Zero seems like the wrong fix for that issue. I'd rather the AMQP library itself be fixed to be more consistent with handling it